app/app.rb

```diff
@@ -0,0 +1,108 @@
+require 'sinatra'
+require 'sinatra/base'
+require 'sinatra/flash'
+require './app/data_mapper_setup'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  register Sinatra::Flash
+
+  get '/' do
+    if current_user
+      redirect '/sessions/new'
+    else
+      @all_users = User.all.map{ |user| user.username }
```

Comment:
Space missing to the left of {.<br>Pass `&:username` as an argument to `map` instead of a block.